test/extensions/filters/http/wasm/wasm_filter_test.cc

```diff
@@ -474,19 +474,28 @@ TEST_P(WasmHttpFilterTest, BodyRequestBufferBody) {
       .WillRepeatedly(Invoke([&bufferedBody](BufferFunction f) { f(bufferedBody); }));
 
   Buffer::OwnedImpl data1("hello");
-  bufferedBody.add(data1);
   EXPECT_CALL(filter(), log_(spdlog::level::err, Eq(absl::string_view("onBody hello"))));
   EXPECT_EQ(Http::FilterDataStatus::StopIterationAndBuffer, filter().decodeData(data1, false));
+  // The first call to decodeData() will not result in the addDecodedData() callback being called
+  // because the filter doesn't know will the VM's onRequestBody() return StopIterationAndBuffer or
```

Comment:
"because the filter doesn't yet know if the VM's onRequestBody() will return StopIterationAndBuffer or Continue"